File: src/prime.rb
# gem update --system
# gem install cmath


require 'cmath'
require 'benchmark'

def is_prime(n)
  return false if n <= 1

  end_num = CMath.sqrt(n).to_i
  (2..end_num).each do |i|
    return false if n % i == 0
  end

  true
end

start_time = Benchmark.realtime do
  c = 0
  (0...9_000_000).each do |i|
    c += 1 if is_prime(i)
  end
  puts c
end

puts start_time * 1000

File: src/prime.pl
#!/usr/bin/perl -l

use Time::HiRes;


sub is_prime {
    $n = $_[0];

    if( $n <= 1 ) {
        return 0
    }
    $end = int(sqrt($n));

    for( $a = 2; $a < $end + 1; $a = $a + 1 ) {
        if ( $n % $a == 0 ) {
            return 0
        }
    }
    return 1
}

$start_time = Time::HiRes::gettimeofday();

$c = 0;

for( $i = 0; $i < 9000000; $i = $i + 1 ) {
    if ( is_prime( $i ) ) {
        $c = $c + 1;
    }
}

print $c;
print Time::HiRes::gettimeofday() - $start_time;
